smarts/core/agent_interface.py

```diff
@@ -194,17 +196,25 @@ class AgentsAliveDoneCriteria:
 
 
 @dataclass(frozen=True)
-class ActorsAliveDoneCriteria:
-    """Require actors to persist."""
+class InterestDoneCriteria:
+    """Require scenario marked interest actors to exist."""
 
-    actors_of_interest: Tuple[str, ...] = ()
-    """Actors that should exist to continue this agent."""
+    actors_filter: Tuple[str, ...] = ()
+    """Interface defined interest actors that should exist to continue this agent."""
 
-    strict: bool = True
+    include_scenario_marked: bool = True
+    """If scenario marked interest actors should be considered as interest vehicles."""
+
+    strict: bool = False
     """If strict the agent will be done instantly if an actor of interest is not available
     immediately.
     """
 
+    @cached_property
+    def actors_pattern(self) -> re.Pattern:
+        """The expression match pattern for actors covered by this interface specifically."""
+        return re.compile("|".join(rf"(?:{aoi})" for aoi in self.actors_filter))
+
```

Comment:
Would it be better to place this method elsewhere?

Reply:
Placing it elsewhere would complicate its use. It is fine since it has no side effects.

My consideration is if I should compress `actors_pattern` and `actors_filter` to be a single attribute since it is not clear that `actors_filter` uses regular expression matching.